linode_api4/objects/linode.py

```diff
@@ -1846,6 +1886,182 @@ def stats_for(self, dt):
             model=self,
         )
 
+    def interface_create(
+        self,
+        firewall: Optional[Union[Firewall, int]] = None,
+        default_route: Optional[
+            Union[Dict[str, Any], LinodeInterfaceDefaultRouteOptions]
+        ] = None,
+        public: Optional[
+            Union[Dict[str, Any], LinodeInterfacePublicOptions]
+        ] = None,
+        vlan: Optional[
+            Union[Dict[str, Any], LinodeInterfaceVLANOptions]
+        ] = None,
+        vpc: Optional[Union[Dict[str, Any], LinodeInterfaceVPCOptions]] = None,
+        **kwargs,
+    ) -> LinodeInterface:
+        """
+        Creates a new interface under this Linode.
+
+        API Documentation: https://techdocs.akamai.com/linode-api/reference/post-linode-interface
+
+        Example: Creating a simple public interface for this Linode::
+
+            interface = instance.interface_create(
+                default_route=LinodeInterfaceDefaultRouteOptions(
+                    ipv4=True,
+                    ipv6
+                ),
+                public=LinodeInterfacePublicOptions()
+            )
+
+        :param firewall: The firewall this interface should be assigned to.
+        :param default_route: The desired default route configuration of the new interface.
+        :param public: The public-specific configuration of the new interface.
+                    If set, the new instance will be a public interface.
+        :param vlan: The VLAN-specific configuration of the new interface.
+                  If set, the new instance will be a VLAN interface.
+        :param vpc: The VPC-specific configuration of the new interface.
+                    If set, the new instance will be a VPC interface.
+
+        :returns: The newly created Linode Interface.
+        """
+
+        params = {
+            "firewall_id": firewall,
```

Comment:
The API complains about the `firewall_id` key if it's fully excluded from the request, but doesn't complain when the key if explicitly specified as null:

```
[400] firewall_id: No default firewall has been defined for this interface type. Please define a default firewall or provide an ID (or 'null') for a specific firewall.
```

Does anyone know if this was an intentional API decision or if it's just a bug?